### DISCUSSION TITLE
A possible bug in Osmand

### DISCUSSION THREAD

**Discussion by laramaceo** [osmandapp/OsmAnd#19099] [General]
Hello!

I am Osmand user and Open street map contributor. I am not completely sure, but I think I have found a bug in Osmand. I have placed an artwork as a point of interest and added to the label the name of the artist and the architect of the artwork. I see that above the artist's name the title "Artist" appears, but the title "architect" does not appear above the name of the work's architect. If this is a bug in the application, could you fix it? I hope I am understood, this is a Spanish language translation and I hope the fix is available for the Spanish language as well. I attach two illustrative images. Thank you very much.

![Missing the title architect](https://github.com/osmandapp/OsmAnd/assets/79804221/9459cb40-762e-45cf-acdc-851ac1dac7ba)

![Tag data](https://github.com/osmandapp/OsmAnd/assets/79804221/d4e6e8fc-d47c-4cd6-ba70-cca2acd61261)

**Comment by osmandlover** [comment]
Hi laramaceo, I took a look at your issue and I can explain why this is happening. 

On line 360 of the `OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityUIHelper.java` file, OsmAnd blanks out the title (which it calls the "textPrefix") of any text label that has an icon, except for the description. 

```
	if (!isDescription && icon == null) {
		icon = getRowIcon(view.getContext(), pType.getIconKeyName());
		if (isText && icon != null) {
			textPrefix = "";
		}
	}
```
You will notice in your screenshot you posted that "Architect" has a person icon, but "Artist name" doesn't, it just has a generic (i) icon.

I think the idea here is that items that have an icon are sufficient to explain what they are. If you look at the calendar one for example, that should tell you that it is a date, and so perhaps the tag "Start date" isn't needed. The ones that are numbers (e.g. height, width) have both labels and icons, and the ones that are enums (a few possible values, e.g. Artwork type, material) have a label as part of the message. 

The easy way to fix this is to simply remove the code I quoted completely, which makes every line item have a label like this:

![image](https://github.com/osmandapp/OsmAnd/assets/142730423/0aadcaa5-87d5-4ae4-ac8b-533c4d87ccfc)

However I have made a dirtier but safer fix that just adds the title for the Architect field specifically. This has a side effect of changing the icon slightly to a specific one for architect, I have no idea why, but I think the specific architect icon is better. The solution looks like this:

![image](https://github.com/osmandapp/OsmAnd/assets/142730423/ccf2cd2b-d38d-45fb-9ee2-156860bd1216)

I'll submit a pull request for this and hopefully the OsmAnd maintainers will agree and merge it.

**Reply by osmandlover** [reply]
Pull request submitted here: https://github.com/osmandapp/OsmAnd/pull/19129

**Reply by laramaceo** [reply]
Thank you for the answer and the solution, I hope your change will be accepted.  

![Seagull](https://github.com/osmandapp/OsmAnd/assets/79804221/d1fd7d6a-17f2-4e1e-ac17-65955caa33f6)

Seagull
The image of the sculpture.
